nrf-hal-common/src/i2s.rs

```diff
@@ -259,14 +259,20 @@ impl I2S {
             _ => Channels::Right,
         }
     }
+
     /// Receives data into the given `buffer` until it's filled.
+    /// Buffer address must be 4 byte aligned and located in RAM.
     /// Returns a value that represents the in-progress DMA transfer.
     #[allow(unused_mut)]
     pub fn rx<W, B>(mut self, mut buffer: B) -> Result<Transfer<B>, Error>
     where
+        W: SupportedWordSize,
         B: WriteBuffer<Word = W>,
     {
         let (ptr, len) = unsafe { buffer.write_buffer() };
+        if ptr as u32 % 4 != 0 {
```

Comment:
```suggestion
        if ptr as u32 & 3 != 0 {
```

The compiler will almost certainly do the right thing here on release, probably even on debug, but just to be sure we could change to a operation with less overhead. Same thing for the other parts of the code where you do this checking.

Reply:
Not sure if I'm a fan of that. That's the job of a compiler, after all.

Reply:
It definately makes the code harder to understand... So can we assume the comiler will do it?

Reply:
It definitely will as soon as any optimizations are enabled, and I don't think we should care too much about performance or binary size with optimizations turned off (Rust should generally be expected to have ~Python-like performance in debug mode, so people will turn on optimizations if they want something better)

Reply:
Agree!